lucene/core/src/java/org/apache/lucene/codecs/blocktree/SegmentTermsEnum.java

```diff
@@ -321,6 +321,10 @@ public boolean seekExact(BytesRef target) throws IOException {
       throw new IllegalStateException("terms index was not loaded");
     }
 
+    if (fr.size() > 0 && (target.compareTo(fr.getMin()) < 0 || target.compareTo(fr.getMax()) > 0)) {
```

Comment:
+1
Interestingly, while working on UniformSplit PostingsFormat we identified the value in this -- `org.apache.lucene.codecs.uniformsplit.BlockReader#seekExact(org.apache.lucene.util.BytesRef)` 
If you tend to add data very sequentially in terms of IDs (common for many apps, log data, and especially during full indexing), this can be very noticeable since most segments will rule out the ID completely.

Still, to be sure, we should run https://github.com/mikemccand/luceneutil/